lms/djangoapps/courseware/masquerade.py

```diff
@@ -99,6 +126,34 @@ def get_masquerade_role(user, course_key):
     return course_masquerade.role if course_masquerade else None
 
 
+def get_masquerade_user(user, course_key):
+    """
+    Returns the specific Django user the authenticated user is masquerading as, or None if masquerading as a
+    specific user is not active or invalid.
+    """
+    # TODO(smarnach): We should notify the caller if the user_name in the masquerading settings don't refer
+    # to a valid user enrolled in this course.  This shouldn't normally be possible, since we check whether the
+    # user is enrolled in the AJAX callback, but the enrollment state of a user might have changed in the
+    # meantime.  We could also reset the masquerading settings in that case.
+    course_masquerade = get_course_masquerade(user, course_key)
+    if course_masquerade and course_masquerade.user_name:
+        try:
+            masquerade_user = CourseEnrollment.objects.users_enrolled_in(course_key).get(username=course_masquerade.user_name)
+        except User.DoesNotExist:
+            return None
+        else:
+            # The masquerade_user object needs to carry the masquerading_settings from this point,
+            # since the functions in this module look them up on the user object.  We also remove
+            # the settings from the original user object, since the real_user is only used in places
+            # where we want to circumvent masquerading, e.g. to show the "Staff Debug Info" link
+            # when masquerading as a student.
+            masquerade_user.masquerade_settings = user.masquerade_settings
+            del user.masquerade_settings
+            masquerade_user.real_user = user
+            return masquerade_user
```

Comment:
And I also combined the two setup functions now, since it's not necessary to keep them separate anymore.  This enabled handling the edge case that the user we try to masquerade as has been unenrolled in the meantime by simply resetting the masquerading settings.